plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/Utils.php

```diff
@@ -125,13 +125,19 @@ public static function get_product_from_context( $block, $previous_product ) {
 	}
 
 	/**
-	 * Check if a product is a simple product that is not purchasable or not in stock.
+	 * Check if a product is not purchasable or not in stock.
 	 *
 	 * @param \WC_Product $product The product to check.
-	 * @return bool True if the product is a simple product that is not purchasable or not in stock.
+	 * @return bool True if the product is not purchasable or not in stock.
 	 */
-	public static function is_not_purchasable_simple_product( $product ) {
-		return ProductType::SIMPLE === $product->get_type() && ( ! $product->is_in_stock() || ! $product->is_purchasable() );
+	public static function is_not_purchasable_product( $product ) {
+		if ( $product->is_type( 'simple' ) ) {
+			return ! $product->is_in_stock() || ! $product->is_purchasable();
+		} elseif ( $product->is_type( 'variable' ) ) {
+			return ! $product->has_purchasable_variations();
```

Comment:
Conceptual question: Since this is a product-based utility, I'm curious why we don't check the parent's stock status before digging into variations? 🤔

Reply:
Good call, updated in 62ab64360cc26a74c706e1f8a412b27346157578.